src/client/java/daos-java/src/main/native/io_daos_obj_DaosObjClient.c

```diff
@@ -702,16 +783,38 @@ Java_io_daos_obj_DaosObjClient_allocateSimDescGroup(
 	char *buffer = (char *)memAddress;
 	uint64_t address;
 	int i;
+	int rc;
 
+	if (grp == NULL) {
+		throw_exception_const_msg_object(env,
+		"memory allocation failed", CUSTOM_ERR3);
+		return -1L;
+	}
 	grp->descs = (data_desc_simple_t **)malloc(
 		nbr * sizeof(data_desc_simple_t *));
+	if (grp->descs == NULL) {
+		throw_exception_const_msg_object(env,
+		"memory allocation failed", CUSTOM_ERR3);
+		return -1L;
+	}
 	grp->nbrOfDescs = nbr;
 	for (i = 0; i < nbr; i++) {
 		grp->descs[i] = (data_desc_simple_t *)malloc(
 			sizeof(data_desc_simple_t));
+		if (grp->descs[i] == NULL) {
+			throw_exception_const_msg_object(env,
+			"memory allocation failed", CUSTOM_ERR3);
+			return -1L;
+		}
 		memcpy(&address, buffer, 8);
 		buffer += 8;
-		allocate_simple_desc((char *)address, grp->descs[i], 1);
+		rc = allocate_simple_desc((char *)address,
+        			grp->descs[i], 1);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs